web-server/src/slices/service.ts

```diff
@@ -0,0 +1,76 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+
+import { ServiceNames } from '@/constants/service';
+
+type Status = {
+  isUp: boolean;
+};
+
+type Service = {
+  isUp: boolean;
+  logs: string[];
+};
+
+export type ServiceStatusState = {
+  [key in ServiceNames]: Service;
+};
+
+type State = {
+  services: ServiceStatusState;
+  loading: boolean;
+  error?: string;
+  active: string | null;
```

Comment:
not string, 

saw it being implemented like 
`const active = useSelector((s) => s.service.active) as ServiceNames;`

so initialize it as 
```ts
active: ServiceNames | null;
```

Reply:
then use can remove typecasting from here:
`const active = useSelector((s) => s.service.active)`

Reply:
I am removing this state to minimize the rerender I will pass the active service name as prop to the overlay page